File: src/routes/public/auth.routes.js
import express from "express"
import dotenv from "dotenv"

import { expressSessionMiddleware } from "../../middlewares/auth.middlewares.js"
import * as signinControllers from "../../controllers/auth/signin.controllers.js"
import * as passwordResetControllers from "../../controllers/auth/passwordReset.controllers.js"
import * as signupControllers from "../../controllers/auth/signup.controllers.js"
import * as signupValidators from "../../validators/auth/signup.validators.js"
import * as signinValidators from "../../validators/auth/signin.validators.js"
import * as passwordResetValidators from "../../validators/auth/passwordReset.validators.js"

dotenv.config()

const router = express.Router()

router.use(
  "/signup",
  expressSessionMiddleware(
    "ongoing_registration",
    process.env.SIGNUP_SESSION_COOKIE_SECRET,
    "/api/auth/signup"
  )
)

router.use(
  "/forgot_password",
  expressSessionMiddleware(
    "ongoing_password_reset",
    process.env.PASSWORD_RESET_SESSION_COOKIE_SECRET,
    "/api/auth/forgot_password"
  )
)

router.post(
  "/signup/request_new_account",
  ...signupValidators.requestNewAccount,
  signupControllers.requestNewAccount
)
router.post(
  "/signup/verify_email",
  ...signupValidators.verifyEmail,
  signupControllers.verifyEmail
)
router.post(
  "/signup/register_user",
  ...signupValidators.registerUser,
  signupControllers.registerUser
)

router.post("/signin", ...signinValidators.signin, signinControllers.signin)

router.post(
  "/forgot_password/request_password_reset",
  ...passwordResetValidators.requestPasswordReset,
  passwordResetControllers.requestPasswordReset
)
router.post(
  "/forgot_password/confirm_email",
  ...passwordResetValidators.confirmEmail,
  passwordResetControllers.confirmEmail
)
router.post(
  "/forgot_password/reset_password",
  ...passwordResetValidators.resetPassword,
  passwordResetControllers.resetPassword
)

export default router

File: src/routes/authRoutes/authRoutes.go
package authRoutes

import (
	"i9lyfe/src/controllers/auth/passwordResetControllers"
	"i9lyfe/src/controllers/auth/siginControllers"
	"i9lyfe/src/controllers/auth/signupControllers"

	"github.com/gofiber/fiber/v2"
)

func Init(router fiber.Router) {
	router.Post("/signup/request_new_account", signupControllers.RequestNewAccount)
	router.Post("/signup/verify_email", signupControllers.VerifyEmail)

	router.Post("/signup/register_user", signupControllers.RegisterUser)

	router.Post("/signin", signinControllers.Signin)

	router.Post("/forgot_password/request_password_reset", passwordResetControllers.RequestPasswordReset)

	router.Post("/forgot_password/confirm_action", passwordResetControllers.ConfirmAction)

	router.Post("/forgot_password/reset_password", passwordResetControllers.ResetPassword)
}
